src/Preprocessing.py

```
@@ -0,0 +1,67 @@
+
+def auto_rotate_if_landscape(img):
+    h, w = img.shape[:2]
+    if w > h:   #ngang → dọc
+        img = cv2.rotate(img, cv2.ROTATE_90_CLOCKWISE)
+    return img
+
+# Xoay ảnh theo EXIF (nếu có)
+def load_and_correct_orientation(image_path):
+    img = Image.open(image_path)
+    try:
+        for orientation in ExifTags.TAGS.keys():
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)